homeassistant/components/synology_dsm/coordinator.py

```diff
@@ -0,0 +1,148 @@
+        entry: ConfigEntry,
+        api: SynoApi,
+        update_interval: timedelta,
+    ) -> None:
+        """Initialize synology_dsm DataUpdateCoordinator."""
+        self.api = api
+        self.entry = entry
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{entry.title} {self.__class__.__name__}",
+            update_interval=update_interval,
+        )
+
+
+class SynologyDSMSwitchUpdateCoordinator(SynologyDSMUpdateCoordinator):
+    """DataUpdateCoordinator to gather data for a synology_dsm switch devices."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        api: SynoApi,
+    ) -> None:
+        """Initialize DataUpdateCoordinator for switch devices."""
+        super().__init__(hass, entry, api, timedelta(seconds=30))
+        self.version: str | None = None
+
+    async def async_setup(self) -> None:
+        """Set up the coordinator initial data."""
+        info = await self.hass.async_add_executor_job(
+            self.api.dsm.surveillance_station.get_info
+        )
+        self.version = info["data"]["CMSMinVersion"]
+
+    async def _async_update_data(self) -> dict[str, dict[str, SynoCamera]] | None:
+        """Fetch all data from api."""
+        surveillance_station = self.api.surveillance_station
+        return {
+            "switches": {
+                "home_mode": await self.hass.async_add_executor_job(
+                    surveillance_station.get_home_mode_status
+                )
+            }
+        }
+
+
+class SynologyDSMCentralUpdateCoordinator(SynologyDSMUpdateCoordinator):
+    """DataUpdateCoordinator to gather data for a synology_dsm central device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        api: SynoApi,
+    ) -> None:
+        """Initialize DataUpdateCoordinator for central device."""
+        super().__init__(
+            hass,
+            entry,
+            api,
+            timedelta(
+                minutes=entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+            ),
+        )
+
+    async def _async_update_data(self) -> dict[str, dict[str, SynoCamera]] | None:
+        """Fetch all data from api."""
+        try:
+            await self.api.async_update()
+        except SYNOLOGY_CONNECTION_EXCEPTIONS as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+        return None
+
+
+class SynologyDSMCameraUpdateCoordinator(SynologyDSMUpdateCoordinator):
+    """DataUpdateCoordinator to gather data for a synology_dsm cameras."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        api: SynoApi,
+    ) -> None:
+        """Initialize DataUpdateCoordinator for cameras."""
+        super().__init__(hass, entry, api, timedelta(seconds=30))
+
+    async def _async_update_data(self) -> dict[str, dict[str, SynoCamera]] | None:
```

Comment:
It never returns `None`?

Reply:
Oh. I forgot to adjust this when I took out all the guards that were no longer needed.